### ISSUE TITLE
[BUG] Duplicating information in trakt_cache

### ISSUE THREAD

**Issue by JourneyOver** [iamkroot/trakt-scrobbler#109]
**Describe the bug**

Haven't updated trakt scrobber in a good while, and decided to check things out and seen that there was an update so updated.. Now whenever I play a show that was previously in my cache before, it seems to add a duplicate entry into the cache but with the year tact on the end.

Example:

I already have:
`"Marvel's Jessica Jones": 79385` 
in the cache from previous plays before I updated, now upon playing an episode of `Marvel's Jessica Jones` it adds a duplicate entry into the cache as `"Marvel's Jessica Jones2015": 79385` so now there are two separate entries in the cache for the same show and trakt ID.

```
"Marvel's Jessica Jones": 79385,
"Marvel's Jessica Jones2015": 79385 
```

**Desktop (please complete the following information):**
 - OS and Version: Windows 10 Latest Updates
 - Python Version: 3.8.3
 - Player and Version: Plex (latest updates)

Downgrading down to 1.2.0b1 seems to stop the issue which is weird because I believe I was previously on 1.2.0b2 but I was pulling from github instead, so maybe I was on 1.2.0b2 but before you pushed 1.2.0b2 into full frutition (I think it was this commit https://github.com/iamkroot/trakt-scrobbler/commit/362abaf36c53eb0f67b08403356332227c83a2f3 here that started possibly adding it, while I was on the commit right before it).

**Comment by iamkroot** [comment]
Hey @JourneyOver, this isn't really a bug. Here's what happens when we see a new show:
1. We look in the cache file to see if we already know the TraktID of this particular media. If so, use that.
2. If not, we need to do a search on trakt to find out the Trakt ID.
3. Once we get the ID, we store it in cache, so that step 1 will succeed the next time same show/movie is scrobbled.

In short, the cache file exists so that we don't make an extra API call (to identify the show/movie ID) on every single scrobble. The scrobbler can actually work perfectly fine without having that cache, at the expense of making an extra request every time.

Coming to the year field, earlier, we stored the trakt id for every show name, but turns out, names aren't unique! That's why, when searching for the trakt ID (in step 2), we started using year (see #66), since we can be pretty sure there will be only one show with the given name in a particular year.

But in #66, I forgot to modify the cache behaviour. It would still store the mapping for name->traktID. For example, the scrobbler couldn't differentiate between "Mulan (1998)" and "Mulan (2020)", since in step 1, it only looked for "Mulan" in the cache.

This was fixed in https://github.com/iamkroot/trakt-scrobbler/commit/362abaf36c53eb0f67b08403356332227c83a2f3 as you noticed, by adding year to the key for the mapping in the cache. Now, during step 1, the scrobbler looks for "Mulan2020" (assuming the file name contains the year; if not, it will still look for "Mulan", since we don't know the year).

Particularly in your case, after https://github.com/iamkroot/trakt-scrobbler/commit/362abaf36c53eb0f67b08403356332227c83a2f3, the scrobbler started looking for `Marvel's Jessica Jones2015` in the cache, but it actually failed in step 1 for the first time, since the cache only had `Marvel's Jessica Jones`. So it must have done the next steps, and this time, in step 3, it stored Title+Year.

**TL;DR**
Simply put, the older entries without the year are redundant, and you can delete those lines safely.

**Comment by JourneyOver** [comment]
Hmm okay makes sense.

Instead of deleting the old entries without the year, do you think it would be better to just go in and add the year into the old entries like how it would do so with said new entries? That way it could skip having to do the steps again to get the trakt ID and all that for all the old content?

**Comment by iamkroot** [comment]
You could do that, but I really don't recommend it. We want to avoid making an extra request for every scrobble, but making one request per show is pretty cheap. You should let the scrobbler populate the cache on its own.

Another aspect is that many of the older entries are simply not needed anymore - you completed the movie/show and are unlikely to watch it again for quite some time. In this case, you'd be wasting your own time by adding years to those entries. 

Rather, I recommend periodically deleting the entire cache file (once a year should be fine). The scrobbler can recover from it very cheaply. I haven't felt the need to support this automatically, because it is _very_ unlikely to grow beyond a few MBs (over decades), and that isn't something to worry about.
